drivers/flash/nrf_qspi_nor.c

```diff
@@ -461,36 +508,129 @@ static int qspi_nrfx_configure(const struct device *dev)
 	qspi_fill_init_struct(&QSPIconfig);
 
 	nrfx_err_t res = nrfx_qspi_init(&QSPIconfig, qspi_handler, dev_data);
+	int ret = qspi_get_zephyr_ret_code(res);
+
+	if ((ret == 0)
+	    && (INST_0_QER != JESD216_DW15_QER_NONE)) {
+		/* Set QE to match transfer mode.  If not using quad
+		 * it's OK to leave QE set, but doing so prevents use
+		 * of WP#/RESET#/HOLD# which might be useful.
+		 *
+		 * Note build assert above ensures QER is S1B6.  Other
+		 * options require more logic.
+		 */
+
+		ret = qspi_rdsr(dev);
+		if (ret < 0) {
+			LOG_ERR("RDSR failed: %d", ret);
+			return ret;
+		}
 
-	if (res == NRFX_SUCCESS) {
-		/* If quad transfer was chosen - enable it now */
-		if ((qspi_is_used_write_quad_mode(QSPIconfig.prot_if.writeoc))
-		    || (qspi_is_used_read_quad_mode(QSPIconfig.prot_if.readoc))) {
+		uint8_t sr = (uint8_t)ret;
+		bool qe_value =
+			(qspi_write_is_quad(QSPIconfig.prot_if.writeoc))
+			|| (qspi_read_is_quad(QSPIconfig.prot_if.readoc));
+		const uint8_t qe_mask = BIT(6); /* only S1B6 */
```

Comment:
Comment that BIT(6) means QE (Quad Enable) in context of status register would be nice.

Reply:
I'm not really sure how to do that.  `S1B6` means bit 6 of the first byte of the status register, a comment just above notes that we know QER is S1B6, `qe` is right there in the variable name, and this whole block is about setting QE in the status register.   Any more comments I can come up with seem confusing.